implementations/rust/ockam/ockam_identity/src/secure_channel/handshake/handshake_worker.rs

```diff
@@ -93,6 +95,7 @@ impl Worker for HandshakeWorker {
             } else {
                 Err(IdentityError::UnknownChannelMsgDestination.into())
             };
+            self.flag.store(true, Ordering::Relaxed);
```

Comment:
In this case why is this not working? if I put this line outside of this code block It's work normally.

Reply:
I can speculate that the spawned task is checking the flag **before** the initialization completes, whereas if you move the flag.store in the upper block it would also be called during initilization.

Reply:
probably starting from a `true` value would do the trick